### PR TITLE
fix issue with k8s_job_op when combining command and container_config

### PR DESCRIPTION
**Pull request by gibsondan**
Summary:
You can't add a key to an immutable dict.

Test Plan: New test case

### CODE REVIEW

File: python_modules/libraries/dagster-k8s/dagster_k8s/ops/k8s_job_op.py [hunk 1]
```diff
@@ -203,7 +203,7 @@ def execute_k8s_job(
         else None,
     )
 
-    container_config = container_config or {}
+    container_config = container_config.copy() or {}
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-k8s/dagster_k8s/ops/k8s_job_op.py:206]: should we update the mypy type to `Mapping` to indicate it is / can be read only?

**Review by alangenfeld — APPROVED** [review]

File: python_modules/libraries/dagster-k8s/dagster_k8s/ops/k8s_job_op.py (hunk 1 above)
Comment (gibsondan) [reply python_modules/libraries/dagster-k8s/dagster_k8s/ops/k8s_job_op.py:206]: I like this idea but it doesn't like calling .copy() on a Mapping

Comment (gibsondan) [reply python_modules/libraries/dagster-k8s/dagster_k8s/ops/k8s_job_op.py:206]: (mypy doesn't)

Comment (schrockn) [reply python_modules/libraries/dagster-k8s/dagster_k8s/ops/k8s_job_op.py]: To do this with Mapping you can use unpack it using `**`.

`new_dict = {**mapping}`

**Pull request merged**